app/lib/actions/katello/host/hypervisors.rb

```diff
@@ -2,14 +2,23 @@ module Actions
   module Katello
     module Host
       class Hypervisors < Actions::EntryAction
-        def plan(hypervisor_params)
+        def plan(options = {})
+          hypervisor_params = options.fetch(:hypervisor_params, nil)
+          task_id = options.fetch(:task_id, nil)
           sequence do
-            hypervisor_results = ::Katello::Resources::Candlepin::Consumer.register_hypervisors(hypervisor_params)
-            plan_action(Katello::Host::HypervisorsUpdate, parse_hypervisors(hypervisor_results))
+            if task_id
+              task_output = plan_action(Candlepin::AsyncHypervisors, :task_id => task_id).output
+              parsed_response = task_output[:hypervisors]
+            elsif hypervisor_params
+              hypervisor_results = ::Katello::Resources::Candlepin::Consumer.register_hypervisors(hypervisor_params)
+              parsed_response = Hypervisors.parse_hypervisors(hypervisor_results)
+            end
+
+            plan_action(Katello::Host::HypervisorsUpdate, :hypervisors => parsed_response)
           end
         end
 
-        def parse_hypervisors(hypervisor_results)
+        def self.parse_hypervisors(hypervisor_results)
```

Comment:
right now I can take it from input of the other action that is planned (so it's already stored anyway) but it'd be nice to have it as output of this action, since this one generates it